homeassistant/components/music_assistant/entity.py

```diff
@@ -0,0 +1,91 @@
+"""Base entity model."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from music_assistant.common.models.enums import EventType
+from music_assistant.common.models.event import MassEvent
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+if TYPE_CHECKING:
+    from music_assistant.client import MusicAssistantClient
+    from music_assistant.common.models.player import Player
+
+
+class MassBaseEntity(Entity):
+    """Base Entity from Music Assistant Player."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, mass: MusicAssistantClient, player_id: str) -> None:
+        """Initialize MediaPlayer entity."""
+        self.mass = mass
+        self.player_id = player_id
+        self._attr_should_poll = False
+        player = mass.players.get(player_id)
+        if TYPE_CHECKING:
+            assert player is not None
+        provider = self.mass.get_provider(player.provider)
+        if TYPE_CHECKING:
+            assert provider is not None
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, player_id)},
+            manufacturer=player.device_info.manufacturer or provider.name,
+            model=player.device_info.model or player.name,
+            name=player.display_name,
+            configuration_url=f"{mass.server_url}/#/settings/editplayer/{player_id}",
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        await self.async_on_update()
+        self.async_on_remove(
+            self.mass.subscribe(
+                self.__on_mass_update, EventType.PLAYER_UPDATED, self.player_id
+            )
+        )
+        self.async_on_remove(
+            self.mass.subscribe(
+                self.__on_mass_update,
+                EventType.QUEUE_UPDATED,
+            )
+        )
+
+    @property
+    def player(self) -> Player:
+        """Return the Mass Player attached to this HA entity."""
+        player = self.mass.players.get(self.player_id)
+        assert player is not None
+        return player
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return unique id for entity."""
+        _base = f"mass_{self.player_id}"
+        if hasattr(self, "entity_description"):
+            return f"{_base}_{self.entity_description.key}"
+        return _base
```

Comment:
I personally always prefer to have this logic in the platforms instead

Reply:
Does this stay where it is?

Reply:
In my opinion this belongs here in the base entity - it can be overridden in a platform if it needs platform specific code (which i highly doubt)